test/common/config/grpc_mux_impl_test.cc

```diff
@@ -148,19 +158,21 @@ TEST_F(GrpcMuxImplTest, TypeUrlMismatch) {
   }
 
   {
-    std::unique_ptr<envoy::api::v2::DiscoveryResponse> response(
-        new envoy::api::v2::DiscoveryResponse());
-    response->set_type_url("foo");
-    response->mutable_resources()->Add()->set_type_url("bar");
+    invalid_response->set_type_url("foo");
+    invalid_response->mutable_resources()->Add()->set_type_url("bar");
     EXPECT_CALL(callbacks_, onConfigUpdateFailed(_)).WillOnce(Invoke([](const EnvoyException* e) {
       EXPECT_TRUE(
           IsSubstring("", "", "bar does not match foo type URL is DiscoveryResponse", e->what()));
     }));
-    expectSendMessage("foo", {"x", "y"}, "");
-    grpc_mux_->onReceiveMessage(std::move(response));
-  }
 
-  expectSendMessage("foo", {}, "");
+    expectSendMessage("foo", {"x", "y"}, "", Grpc::Status::GrpcStatus::Internal,
+                      fmt::format("{} does not match {} type URL is DiscoveryResponse {}", "bar",
+                                  "foo", invalid_response->DebugString()));
+    grpc_mux_->onReceiveMessage(std::move(invalid_response));
+  }
+  expectSendMessage("foo", {}, "", Grpc::Status::GrpcStatus::Internal,
+                    fmt::format("{} does not match {} type URL is DiscoveryResponse {}", "bar",
```

Comment:
So, we latch the error message for all subsequent requests? This seems like it might be a bit confusing.

Reply:
@htuch no we do not actually. I am clearing it on the next onReceiveMessage. Will look in to it.

Reply:
I found the issue. The actual clear_error_detail call should be immediately after request is sent in  sendRequest rather than in onReceiveMessage.